packages/core/src/babel-plugin-adjust-imports.ts

```diff
@@ -353,6 +354,24 @@ function makeExternal(specifier: string, sourceFile: AdjustFile, opts: Options):
   return explicitRelative(dirname(sourceFile.name), target.slice(0, -3));
 }
 
+function atomicWrite(path: string, content: string) {
+  if (pathExistsSync(path)) {
+    return;
+  }
+  let suffix = randomBytes(8).toString('hex');
+  outputFileSync(path + suffix, content);
+  try {
+    renameSync(path + suffix, path);
+  } catch (err: any) {
+    // windows throws EPERM for concurrent access. For us it's not an error
+    // condition because the other thread is writing the exact same value we
+    // would have.
```

Comment:
Thinking about async func here with few retry attempts. I think it’s important to do job, instead of fail

Reply:
If we hit EPERM here it's only because another process is already doing the exact same job we were trying to do, so we can move on and let that process take care of it.

As far as I can tell, that's the only way to get EPERM here. Since this is inside our own temp dir, you shouldn't ever have unrelated permission failures in the usual Unix EPERM sense.